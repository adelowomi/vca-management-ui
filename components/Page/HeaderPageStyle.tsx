import * as React from 'react';

import { ErrorProps } from '../../types/interfaces';
import {
  ColumnSection,
  H2,
  ImageSelectBox,
} from './PageStyledElements';
import SelectMediaModal from './SelectMediaModal';

export interface PageHeaderStyleProps {
  onButtonClick: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  headerType: string;
  state: any;
  setState: any;
  medias: any;
  handleSubmit?: any;
  errors?: ErrorProps;
  profile;
  token;
}

export function PageHeaderStyle({
  state,
  setState,
  errors,
  profile,
  token,
}: PageHeaderStyleProps): JSX.Element {
  const [open, setOpen] = React.useState(false);
  const setMedia = (media: any) => {
    console.error(media);
    setState({ ...state, media: media.id, mediaUrl: media.image.large });
  };
  return (
    <>
      <ColumnSection>
        <H2 className="mt-6">Media Image</H2>
        <ImageSelectBox
          onClick={() => setOpen(!open)}
          className="mt-5 w-96 flex items-center justify-center cursor-pointer"
        >
          <p>+ Select from media gallery</p>
        </ImageSelectBox>
        {errors && errors.mediaUrl && (
          <span className="text-red-500 mt-1">{errors.mediaUrl}</span>
        )}
        <SelectMediaModal
          open={open}
          close={setOpen}
          profile={profile}
          selected={state.media}
          setMedia={setMedia}
          token={token}
          type={"IMAGE"}
        />
      </ColumnSection>
    </>
  );
}
