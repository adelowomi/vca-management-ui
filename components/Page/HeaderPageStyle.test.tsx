import React from 'react';

import { fireEvent, render, screen } from '../../test/testUtils';
import { PageHeaderStyle } from './HeaderPageStyle';

const medias = [
  {
    __typename: 'Media',
    id: '60a465d7505c850015bfb114',
    type: 'IMAGE',
    name: 'media name',
    description: 'media description',
    document: null,
    video: null,
    image: {
      __typename: 'MediaImage',
      small:
        'https://vca-documents.s3.ca-central-1.amazonaws.com/87/4936882cac4e578961a255d47456e4/floriane-vita-FyD3OWBuXnY-unsplash-1.png',
      medium:
        'https://vca-documents.s3.ca-central-1.amazonaws.com/3c/ca8c227f9e49179262f195a2d2774d/floriane-vita-FyD3OWBuXnY-unsplash-1.png',
      large:
        'https://vca-documents.s3.ca-central-1.amazonaws.com/e9/1fb74da6fe477494564bc850e3e80a/floriane-vita-FyD3OWBuXnY-unsplash-1.png',
      xLarge:
        'https://vca-documents.s3.ca-central-1.amazonaws.com/85/5f727d97d14d28829f3f447e31a07a/floriane-vita-FyD3OWBuXnY-unsplash-1.png',
    },
  },
  {
    __typename: 'Media',
    id: '60a465d7505c850015bfb115',
    type: 'IMAGE',
    name: 'media name',
    description: 'media description',
    document: null,
    video: null,
    image: {
      __typename: 'MediaImage',
      small:
        'https://vca-documents.s3.ca-central-1.amazonaws.com/87/4936882cac4e578961a255d47456e4/floriane-vita-FyD3OWBuXnY-unsplash-1.png',
      medium:
        'https://vca-documents.s3.ca-central-1.amazonaws.com/3c/ca8c227f9e49179262f195a2d2774d/floriane-vita-FyD3OWBuXnY-unsplash-1.png',
      large:
        'https://vca-documents.s3.ca-central-1.amazonaws.com/e9/1fb74da6fe477494564bc850e3e80a/floriane-vita-FyD3OWBuXnY-unsplash-1.png',
      xLarge:
        'https://vca-documents.s3.ca-central-1.amazonaws.com/85/5f727d97d14d28829f3f447e31a07a/floriane-vita-FyD3OWBuXnY-unsplash-1.png',
    },
  },
];

describe('PageHeaderStyle', () => {
  it('should render PageHeaderStyle snapshot', () => {
    const mockOnButtonClick = jest.fn();
    const { asFragment } = render(
      <PageHeaderStyle
        state={{}}
        setState={jest.fn()}
        medias={medias}
        onButtonClick={mockOnButtonClick}
        headerType="headerTypeOne"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render PageHeaderStyle snapshot', () => {
    const mockOnButtonClick = jest.fn();
    render(
      <PageHeaderStyle
        state={{}}
        setState={jest.fn()}
        medias={medias}
        onButtonClick={mockOnButtonClick}
        headerType="headerTypeOne"
      />
    );
    fireEvent.click(screen.getByText('Header Type 1'));
    expect(mockOnButtonClick).toHaveBeenCalledTimes(1);
  });
});
