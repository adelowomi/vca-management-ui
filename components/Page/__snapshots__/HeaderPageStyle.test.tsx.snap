// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeaderStyle should render PageHeaderStyle snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 49.5rem;
}

.c3 {
  background: rgba(24,144,255,0.1);
  border: 1px dashed #1890ff;
  border-radius: 2px;
  padding: 19px;
}

.c3 p {
  font-family: Inter;
  font-style: normal;
  font-weight: bold;
  font-size: 16px;
  line-height: 19px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: center;
  color: #1890ff;
}

.c1 {
  border: 2px solid #1890FF;
  box-sizing: border-box;
  border-radius: 2px;
  background: rgba(24,144,255,0.1);
  padding: 17px 0px;
  color: #1890FF;
  font-family: Inter;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 19px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 11.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
}

.c1:focus {
  outline: none;
}

.c2 {
  border: 1px solid #828282;
  box-sizing: border-box;
  border-radius: 2px;
  background: #fff;
  padding: 17px 0px;
  color: #828282;
  font-family: Inter;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 19px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 11.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
}

.c2:focus {
  outline: none;
}

<div
    class="mt-8
flex
flex-col"
  >
    <h1
      class="mb-6 text-xl font-semibold"
    >
      1. Hero Type
    </h1>
    <div
      class="c0"
    >
      <button
        class="c1"
        data-headertype="headerTypeOne"
        type="button"
        value="headerTypeOne"
      >
        Header Type 1
      </button>
      <button
        class="c2"
        data-headertype="headerTypeTwo"
        type="button"
        value="headerTypeTwo"
      >
        Header Type 2
      </button>
      <button
        class="c2"
        data-headertype="headerTypeThree"
        type="button"
        value="headerTypeThree"
      >
        Header Type 3
      </button>
      <button
        class="c2"
        data-headertype="headerTypeFour"
        type="button"
        value="headerTypeFour"
      >
        Header Type 4
      </button>
    </div>
    <h1
      class="mt-6 text-xl font-semibold"
    >
      Media Image
    </h1>
    <div
      class="c3 mt-5 w-96 flex items-center justify-center cursor-pointer"
    >
      <p>
        + Select from media gallery
      </p>
    </div>
  </div>
</DocumentFragment>
`;
